inc/namespace.php

```diff
@@ -833,9 +833,21 @@ function elasticpress_analyzer_language() : string {
  * and override the default analyzer.
  *
  * @param array $mapping Mapping array.
+ * @param string|null $index The current index this mapping is for.
  * @return array
  */
-function elasticpress_mapping( array $mapping ) : array {
+function elasticpress_mapping( array $mapping, ?string $index = null ) : array {
+	// Derive current mapping type.
+	$indexables = Indexables::factory()->get_all( null, true );
+	preg_match( '/.+-(' . implode( '|', $indexables ) . ')-\d+$/', $index, $matches );
```

Comment:
This should be `preg_quote`'d just in case
```suggestion
	$escaped_indexables = array_map( function ( $item ) {
		return preg_quote( $item, '/' );
	}, $indexables );
	preg_match( '/.+-(' . implode( '|', $escaped_indexables ) . ')-\d+$/', $index, $matches );
```